### PR TITLE
Import FindTaglib.cmake from ECM 5.72.0 and mark Taglib as REQUIRED

### PR DESCRIPTION
**Pull request by a17r**

### CODE REVIEW

**Comment by dschmidt** [comment]
Great, thanks!

**Comment by hugolm84** [comment]
This was a bit fast merge, why cannot we not use the ECM module directly?

File: src/libtomahawk/CMakeLists.txt [hunk 1]
```diff
@@ -533,8 +533,7 @@ TARGET_LINK_LIBRARIES(${TOMAHAWK_LIBRARY} PRIVATE
 )
 
 # TagLib
-TARGET_LINK_LIBRARIES(${TOMAHAWK_LIBRARY} LINK_PUBLIC ${TAGLIB_LIBRARIES})
-TARGET_INCLUDE_DIRECTORIES(${TOMAHAWK_LIBRARY} PUBLIC ${TAGLIB_INCLUDES})
+target_link_libraries(${TOMAHAWK_LIBRARY} LINK_PUBLIC Taglib::Taglib)
```
Comment (hugolm84) [inline src/libtomahawk/CMakeLists.txt:536]: This will no longer work if taglib was compiled and installed locally with zlib enabled.

**Comment by a17r** [comment]
> This was a bit fast merge, why cannot we not use the ECM module directly?

That would mean depending a release from 2020-07-11. No problem for me, but most projects shy away from introducing such hurdles for users of lesser distributions.

File: src/libtomahawk/CMakeLists.txt (hunk 1 above)
Comment (a17r) [reply src/libtomahawk/CMakeLists.txt:536]: Perfectly possible I was missing something there. Just re-add `target_include_directories`?

Comment (hugolm84) [reply src/libtomahawk/CMakeLists.txt:536]: Its actually the ECM module that does not read out the pkg config correctly and will in turn not append `-lz` to the `${Taglib_LIBRARIES}` variable.

I think a simple work around here would be 

```
IF(ZLIB_FOUND)
    target_link_libraries(${TOMAHAWK_LIBRARY} PUBLIC -lz)
endif()
```

Comment (a17r) [reply src/libtomahawk/CMakeLists.txt:536]: That would be a bug in taglib though and fixed in master: https://github.com/taglib/taglib/commit/ef1312d62239f399c40233d76ef3328b8dadf984

Unsure how best to proceed there. We can't depend on a yet-unreleased version of taglib, though basically every distribution has that covered by now by shipping a snapshot of taglib instead. https://repology.org/project/taglib/versions

Comment (a17r) [reply src/libtomahawk/CMakeLists.txt:536]: Here's what Clementine proj are doing: https://github.com/clementine-player/Clementine/blob/master/CMakeLists.txt#L127

Although it is a desperate move, as taglib upstream have not bumped that version yet from `1.11.1`. And that leads to us patching our taglib snapshot to be `1.11.2`...

Comment (a17r) [reply src/libtomahawk/CMakeLists.txt:536]: How about something like:

```
if (NOT TAGLIB_VERSION VERSION_GREATER 1.11.1)
  IF(ZLIB_FOUND)
    target_link_libraries(${TOMAHAWK_LIBRARY} PUBLIC -lz)
  endif()
endif()
```

Comment (hugolm84) [reply src/libtomahawk/CMakeLists.txt:536]: Thanks, that seems like a sane way forward :)

**Pull request merged**